From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Florian Festi <ffesti@redhat.com>
Date: Mon, 10 Oct 2016 11:57:14 +0200
Subject: [PATCH] rpm2archive: return 0 on success instead of stop iteration
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Solves: https://github.com/rpm-software-management/rpm/pull/94/files
Thanks to Neal Gompa (ニール・ゴンパ)<ngompa13@gmail.com> for spotting this
and proposing a solution!
---
 rpm2archive.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/rpm2archive.c b/rpm2archive.c
index 63d64d9..5cbb36a 100644
--- a/rpm2archive.c
+++ b/rpm2archive.c
@@ -161,6 +161,11 @@ static int process_package(rpmts ts, char * filename)
 	    write_file_content(a, buf, fi);
 	}
     }
+    /* End of iteration is not an error */
+    if (rc == RPMERR_ITER_END) {
+	rc = 0;
+    }
+
     _free(hardlink);
 
     Fclose(gzdi);	/* XXX gzdi == fdi */
